tests/lib/nested.sh

```diff
@@ -667,15 +706,19 @@ start_nested_core_vm_unit(){
     wait_for_ssh
 }
 
-get_current_image(){
-    IMAGE_DIR="$(get_image_dir)"
-    echo "$IMAGE_DIR/ubuntu-core-current.img"
+get_current_image_name() {
+    echo "ubuntu-core-current.img"
 }
 
-start_nested_core_vm(){
-    local IMAGE_DIR CURRENT_IMAGE
-    IMAGE_DIR="$(get_image_dir)"
-    CURRENT_IMAGE="$(get_current_image)"
+start_nested_core_vm() {
+    local CURRENT_IMAGE IMAGE_NAME
+    CURRENT_IMAGE="$NESTED_IMAGES_DIR/$(get_current_image_name)"
+
+    # In case there is not a current image to be started but there is a compressed
+    # one, this is uncompressed and used for the current test
+    if [ ! -f "$CURRENT_IMAGE" ] && [ -f "${CURRENT_IMAGE}.xz" ]; then
+        uncompress_image "$(get_current_image_name)"
+    fi
 
     # In case the current image already exists, it needs to be reused and in that
     # case is neither required to copy the base image nor prepare the ssh
```

Comment:
this section would read better if the two equivalent conditions were combined to be something like:

```bash
if [ ! -f "$CURRENT_IMAGE" ]; then
    if [ -f "${CURRENT_IMAGE}.xz" ]; then
      uncompress_image "$(get_current_image_name)"
    else
        IMAGE_NAME="$(get_image_name core)"
        cp "$IMAGE_DIR/$IMAGE_NAME" "$CURRENT_IMAGE"	        uncompress_image "$IMAGE_NAME"
        mv "$NESTED_IMAGES_DIR/$IMAGE_NAME" "$CURRENT_IMAGE"
        # Start the nested core vm
        start_nested_core_vm_unit "$CURRENT_IMAGE"

        prepare_ssh
       ... etc
    fi
else
    # Start the nested core vm
    start_nested_core_vm_unit "$CURRENT_IMAGE"
fi
```

Reply:
I already updated the conditions in this function. I'll try to simplify that in a following PR